src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -134,14 +146,21 @@ static Guid GetSolutionSessionId()
         }
 
         VSTypeScriptVisualStudioProjectWrapper IVsTypeScriptVisualStudioProjectFactory.CreateAndAddToWorkspace(string projectSystemName, string language, string projectFilePath, IVsHierarchy hierarchy, Guid projectGuid)
+        {
+            return _threadingContext.JoinableTaskFactory.Run(() =>
+                ((IVsTypeScriptVisualStudioProjectFactory)this).CreateAndAddToWorkspaceAsync(projectSystemName, language, projectFilePath, hierarchy, projectGuid, CancellationToken.None));
+        }
+
+        async Task<VSTypeScriptVisualStudioProjectWrapper> IVsTypeScriptVisualStudioProjectFactory.CreateAndAddToWorkspaceAsync(
```

Comment:
❕ Types outside the external access layer cannot directly reference types inside the external access layer.

Reply:
The implementation of `IVsTypeScriptVisualStudioProjectFactory` needs to be moved to a new type `VsTypeScriptVisualStudioProjectFactoryAccessor` in the external access layer for VSTypeScript (one folder above the Api folder where the interface is defined).

Reply:
I'd prefer to do that in a followup. This seems to be continuing the existing pattern we have for EA here, just making an new async entrypoint.